File: src/main/java/com/myplaylists/repository/PlaylistRepository.java
package com.myplaylists.repository;

import com.myplaylists.domain.Playlist;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PlaylistRepository extends JpaRepository<Playlist, Long> {
	List<Playlist> findAllByUserId(Long userId);
	List<Playlist> findAllByUserIdAndTitleContaining(Long userId, String title);
	Page<Playlist> findByVisibility(Pageable pageable, boolean visibility);
	Page<Playlist> findByTitleContaining(Pageable pageable, String title);
	Page<Playlist> findByVisibilityAndTitleContaining(Pageable pageable, boolean visibility, String title);
}

File: src/main/kotlin/com/myplaylists/repository/PlaylistRepository.kt
package com.myplaylists.repository

import com.myplaylists.domain.Playlist
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface PlaylistRepository {
    fun findAllByUserId(userId: Long): List<Playlist>
    fun findAllByUserIdAndTitleContaining(userId: Long, title: String): List<Playlist>
    fun findByVisibility(pageable: Pageable, visibility: Boolean): Page<Playlist>
    fun findByTitleContaining(pageable: Pageable, title: String): Page<Playlist>
    fun findByVisibilityAndTitleContaining(pageable: Pageable, visibility: Boolean, title: String): Page<Playlist>
}